plugins/application/loki/pkg/lib/loki.go

```diff
@@ -0,0 +1,51 @@
+package lib
+
+import (
+	"time"
+	"fmt"
+
+	"github.com/infrawatch/apputils/connector"
+	"github.com/infrawatch/sg-core/pkg/data"
+)
+
+type LokiConfig struct {
+	Connection  string `validate:"required"`
+	BatchSize   int
+	MaxWaitTime int
+}
+
+// Creates labels used by Loki. Discards everything, that isn't
+// a string
+func createLabels(rawLabels map[string]interface{}) (map[string]string, error) {
+	result := make(map[string]string)
+	for key, value := range rawLabels {
+		if typedValue, ok := value.(string); ok {
+			result[key] = typedValue
+		}
+	}
+	if len(result) == 0 {
+		return nil, fmt.Errorf("Unable to create log labels")
+	}
+	return result, nil
+}
+
+func CreateLokiLog(log data.Event) (connector.LokiLog, error) {
+	labels, err := createLabels(log.Labels)
+	if err != nil {
+		return connector.LokiLog{}, err
+	}
+
+	msg, ok := log.Annotations["log_message"].(string)
```

Comment:
The only problem with the Loki plugin expecting the "log_message" index in annotations is that handlers have to create messages specifically for storage into loki. I think it's fine for now, but in the future it may be worth either adding a 'message' field to events or creating an event subtype for logs that contains the field

Reply:
Actually, adding Message attribute to the Event is good idea ... we could then send actual tags for error events in labels and use message instead ... maybe even completely skip adding "description" to annotations and send it empty.

Reply:
From logging point of view a Message attribute in Event would be really good.

Reply:
If you think you can handle this change, feel free to add it to this PR ... since it is logging-connected change anyway.

Reply:
I added the Message attribute to the Event and used it in the logs handler and Loki application.